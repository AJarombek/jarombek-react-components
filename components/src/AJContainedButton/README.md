### Overview

Reusable button component with a specific "contained" style.  For a more generic button, see `AJButton`.

**Use cases:**

```javascript
import { AJContainedButton } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJContainedButton disabled={true}>Contained Button</AJContainedButton>
```

### Files

| Filename                   | Description                                                                           |
|----------------------------|---------------------------------------------------------------------------------------|
| `AJContainedButton.js`     | The reusable `AJContainedButton` component.                                           |
| `index.js`                 | Index file which exports the `AJContainedButton` component.                           |
| `styles.js`                | JSS styles for the `AJContainedButton` component.                                     |
