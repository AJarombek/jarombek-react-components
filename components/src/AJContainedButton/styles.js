/**
 * Stylesheet for the AJContainedButton component using JSS.
 * Author: Andrew Jarombek
 * Date: 1/12/2020
 */

export default {
  ajContainedButton: {
    padding: '10px',
    transition: 'background-color 0.25s ease',
    borderRadius: '4px',

    '& button': {
      backgroundColor: 'transparent',
    },
  },
  ajContainedButtonEnabled: {
    backgroundColor: '#bbb',

    '&:active': {
      backgroundColor: '#ccc',
    },

    '& button': {
      color: '#555',
      transition: 'color 0.25s ease',

      '&:active': {
        color: '#222',
      },
    },
  },
  ajContainedButtonDisabled: {
    backgroundColor: '#eee',

    '& button': {
      color: '#999',
    },
  },
};
