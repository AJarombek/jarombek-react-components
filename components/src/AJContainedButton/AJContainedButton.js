/**
 * Reusable contained button component.
 * @author Andrew Jarombek
 * @since 1/12/2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';

import styles from './styles';
import AJBaseButton from '../AJBaseButton/AJBaseButton';

const useStyles = createUseStyles(styles);

/**
 * Component for a button that displays text and is contained within a solid color.
 * @param children The contents of the button.
 * @param onClick An action that occurs when the button is clicked.
 * @param disabled Whether the button is clickable or not.
 * @return {*} React elements representing a contained button.
 */
const AJContainedButton = ({children, onClick, disabled=false}) => {
  const classes = useStyles();

  const enabledClasses = [
    'aj-contained-button',
    classes.ajContainedButton,
    classes.ajContainedButtonEnabled
  ];

  const disabledClasses = [
    'aj-contained-button',
    classes.ajContainedButton,
    classes.ajContainedButtonDisabled
  ];

  return (
    <AJBaseButton
      onClick={onClick}
      disabled={disabled}
      enabledClasses={enabledClasses}
      disabledClasses={disabledClasses}>

      {children}
    </AJBaseButton>
  );
};

AJContainedButton.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};

export default AJContainedButton;
