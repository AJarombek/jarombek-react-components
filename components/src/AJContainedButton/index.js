import AJContainedButton from './AJContainedButton';

export default AJContainedButton;
