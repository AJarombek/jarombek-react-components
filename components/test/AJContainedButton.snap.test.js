/**
 * Snapshot test for the {@link ../../components/AJContainedButton} component.
 * @author Andrew Jarombek
 * @since 1/13/2020
 */

import React from 'react';
import AJContainedButton from '../src/AJContainedButton/AJContainedButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<AJContainedButton children={<div>Snapshot Test</div>}/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with all props', () => {
  const tree = renderer.create(
    <AJContainedButton
      children={<div>Snapshot Test</div>}
      onClick={f=>f}
      disabled={true}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
