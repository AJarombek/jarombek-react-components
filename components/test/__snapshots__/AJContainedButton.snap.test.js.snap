// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajBaseButton-0-2-4 ajBaseButtonEnabled-0-2-5 aj-contained-button ajContainedButton-0-2-1 ajContainedButtonEnabled-0-2-2"
>
  <button
    disabled={false}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;

exports[`renders correctly with all props 1`] = `
<div
  className="ajBaseButton-0-2-4 ajBaseButtonDisabled-0-2-6 aj-contained-button ajContainedButton-0-2-1 ajContainedButtonDisabled-0-2-3"
  onClick={null}
>
  <button
    disabled={true}
    type="button"
  >
    <div>
      Snapshot Test
    </div>
  </button>
</div>
`;
